google/cloud/spanner/value.h

```diff
@@ -136,7 +137,13 @@ std::pair<google::spanner::v1::Type, google::protobuf::Value> ToProto(Value v);
  */
 class Value {
  public:
-  Value() = delete;
+  /**
+   * Constructs a non-null `Value` that holds nothing.
+   *
+   * All calls to `is<T>()`, `is_null<T>()`, and `get<T>()` will return false
+   * or an error as appropriate.
+   */
+  Value() = default;
```

Comment:
I would consider

`type_.set_code(google::spanner::v1::TypeCode::TYPE_CODE_UNSPECIFIED);`

here, even if it may be the default.